File: src/test/java/com/back/global/rq/TestRq.java
package com.back.global.rq;

import com.back.domain.member.member.entity.Member;
import com.back.domain.member.member.service.MemberService;
import com.back.global.rq.Rq;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import static org.mockito.Mockito.mock;

public class TestRq extends Rq {
    private Member actor;

    public TestRq() {
        // 기존 null에서 Mock 객체로 변경
        super(mock(HttpServletRequest.class), mock(HttpServletResponse.class), mock(MemberService.class));
    }

    public void setActor(Member actor) {
        this.actor = actor;
    }

    @Override
    public Member getActor() {
        return actor;
    }

    @Override
    public String getHeader(String name, String defaultValue) { return defaultValue; }
    @Override
    public void setHeader(String name, String value) { }
    @Override
    public String getCookieValue(String name, String defaultValue) { return defaultValue; }
    @Override
    public void setCrossDomainCookie(String name, String value, int maxAge) { }
    @Override
    public void deleteCrossDomainCookie(String name) { }
}

File: src/test/java/com/back/global/rq/TestRq.kt
package com.back.global.rq

import com.back.domain.member.member.entity.Member
import com.back.domain.member.member.service.MemberService
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.mockito.Mockito.mock

class TestRq : Rq(
    mock(HttpServletRequest::class.java),
    mock(HttpServletResponse::class.java),
    mock(MemberService::class.java)
) {
    private var _actor: Member? = null

    fun setActor(actor: Member) {
        _actor = actor
    }

    override val actor: Member?
        get() = _actor

    override fun getHeader(name: String, defaultValue: String): String = defaultValue
    override fun setHeader(name: String, value: String?) { /* do nothing */ }
    override fun getCookieValue(name: String, defaultValue: String): String = defaultValue
    override fun setCrossDomainCookie(name: String, value: String, maxAge: Int) { /* do nothing */ }
    override fun deleteCrossDomainCookie(name: String) { /* do nothing */ }
}
